source/ftrack_connect_pipeline_3dsmax/asset/dcc_object.py

```diff
@@ -0,0 +1,156 @@
+        self.name = dcc_object
+        return self.name
+
+    def _name_exists(self, name):
+        '''
+        Return true if the given *name* exists in the scene.
+        '''
+
+        # if cmds.objExists(name):
+        #    return True
+
+        return False
+
+    def from_asset_info_id(self, asset_info_id):
+        '''
+        Checks max scene to get all the ftrackAssetNode objects. Compares them
+        with the given *asset_info_id* and returns them if matches.
+        '''
+        ftrack_asset_nodes = max_utils.get_ftrack_nodes()
+        for dcc_object_name in ftrack_asset_nodes:
+
+            # id_value = cmds.getAttr(
+            #     '{}.{}'.format(dcc_object_name, asset_const.ASSET_INFO_ID)
+            # )
+
+            if id_value == asset_info_id:
+                self.logger.debug(
+                    'Found existing object: {}'.format(dcc_object_name)
+                )
+                self.name = dcc_object_name
+                return self.name
+
+        self.logger.debug(
+            "Couldn't found an existing object for the asset info id: {}".format(
+                asset_info_id
+            )
+        )
+        return None
+
+    @staticmethod
+    def dictionary_from_object(object_name):
+        '''
+        Static method to be used without initializing the current class.
+        Returns a dictionary with the keys and values of the given
+        *object_name* if exists.
+
+        *object_name* ftrackAssetNode object type from max scene.
+        '''
+        logger = logging.getLogger(
+            '{0}.{1}'.format(__name__, __class__.__name__)
+        )
+        param_dict = {}
+        # if not cmds.objExists(object_name):
+        #    error_message = "{} Object doesn't exists".format(object_name)
+        #    logger.error(error_message)
+        #    return param_dict
+        # all_attr = cmds.listAttr(object_name, c=True, se=True)
+        for attr in all_attr:
+            # if cmds.attributeQuery(attr, node=object_name, msg=True):
+            #    continue
+            # attr_value = cmds.getAttr('{}.{}'.format(object_name, attr))
+            param_dict[attr] = attr_value
+        return param_dict
+
+    def connect_objects(self, objects):
+        '''
+        Link the given *objects* ftrack attribute to the self
+        :obj:`name` object asset_link attribute in max.
+
+        *objects* List of Max DAG objects
+        '''
+        # for obj in objects:
+        # if cmds.lockNode(obj, q=True)[0]:
+        #     cmds.lockNode(obj, l=False)
+        #
+        # if not cmds.attributeQuery('ftrack', n=obj, exists=True):
+        #     cmds.addAttr(obj, ln='ftrack', at='message')
+        #
+        # if not cmds.listConnections('{}.ftrack'.format(obj)):
+        #     cmds.connectAttr(
+        #         '{}.{}'.format(self.name, asset_const.ASSET_LINK),
+        #         '{}.ftrack'.format(obj),
+        #     )
```

Comment:
Please review this file

Reply:
Dcc object will be attended in a future PR